tools/isobuild/compiler-plugin.js

```diff
@@ -487,6 +530,46 @@ class PackageSourceBatch {
       }
       self.resourceSlots.push(new ResourceSlot(resource, sourceProcessor, self));
     });
+
+    // Compute imports by merging the exports of all of the packages we
+    // use. Note that in the case of conflicting symbols, later packages get
+    // precedence.
+    //
+    // We don't get imports from unordered dependencies (since they may not be
+    // defined yet) or from weak/debugOnly dependencies (because the meaning of
+    // a name shouldn't be affected by the non-local decision of whether or not
+    // an unrelated package in the target depends on something).
+    self.importedSymbolToPackageName = {}; // map from symbol to supplying package name
+    self.usedPackageNames = {};
+
+    compiler.eachUsedUnibuild({
+      dependencies: self.unibuild.uses,
+      arch: self.processor.arch,
+      isopackCache: self.processor.isopackCache,
+      skipUnordered: true,
+      // don't import symbols from debugOnly and prodOnly packages, because
+      // if the package is not linked it will cause a runtime error.
+      // the code must access them with `Package["my-package"].MySymbol`.
+      skipDebugOnly: true,
+      skipProdOnly: true,
+      // We only care about getting exports here, so it's OK if we get the Mac
+      // version when we're bundling for Linux.
+      allowWrongPlatform: true,
+    }, depUnibuild => {
+      _.each(depUnibuild.declaredExports, function (symbol) {
+        // Slightly hacky implementation of test-only exports.
+        if (! symbol.testOnly || self.unibuild.pkg.isTest) {
+          self.importedSymbolToPackageName[symbol.name] = depUnibuild.pkg.name;
+        }
+      });
+
+      self.usedPackageNames[depUnibuild.pkg.name] = true;
+    });
+  }
+
+  usesModules() {
+    return _.isString(this.sourceRoot) &&
```

Comment:
There are still some cases where we don't have a reliable `sourceRoot`, so this property will be `undefined`, such as when we're bundling a plugin.